src/data-generator/index.ts

```diff
@@ -651,91 +657,99 @@ function birthDeclarationWorkflow(
           submissionTime,
           randomFacility
         )
-        const declaration = await fetchRegistration(randomRegistrar, id)
-        try {
-          registrationDetails = await createBirthRegistrationDetailsForNotification(
-            add(new Date(submissionTime), {
-              days: 1
-            }),
-            location,
-            declaration
-          )
-        } catch (error) {
-          console.log(error)
-          console.log(JSON.stringify(declaration))
-          throw error
+        if (REGISTER && CERTIFY) {
```

Comment:
If someone wants to proceed till `CERTIFY` i.e. only uses `CERTIFY=false` then the declaration will not be downloaded from here and down the line when trying to reject the declaration, it should throw an error as the declaration was not downloaded.